addons/web/static/tests/views/fields/boolean_favorite_field_tests.js

```diff
@@ -74,6 +74,46 @@ QUnit.module("Fields", (hooks) => {
         );
     });
 
+    QUnit.test("FavoriteField respects force_save attribute", async function (assert) {
+        await makeView({
+            type: "kanban",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <kanban>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div>
+                                <field name="bar" widget="boolean_favorite" />
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            mockRPC(route, args) {
+                if (args.method === "write" && args.model === "partner") {
+                    assert.step("save");
+                    assert.deepEqual(args.args, [[1], { bar: false }]);
+                }
+            },
+            domain: [["id", "=", 1]],
+        });
+
+        // click on favorite
+        await click(target, ".o_field_widget .o_favorite");
+        assert.containsNone(
+            target,
+            ".o_kanban_record  .o_field_widget .o_favorite > a i.fa.fa-star",
+            "should not be favorite"
+        );
+        assert.strictEqual(
+            target.querySelector(".o_kanban_record .o_field_widget .o_favorite > a").textContent,
+            " Add to Favorites",
+            'the label should say "Add to Favorites"'
+        );
+
+        assert.verifySteps(["save"]);
+    });
```

Comment:
I'd add another test where you set force_save to true (same scenario as this one) and you assert that it doesn't save